lib/bank_accounts/money_market_account.rb

```diff
@@ -0,0 +1,141 @@
+module BankAccounts
+  class MoneyMarketAccount < Account
+    ##--------------------------------------------------------------------------
+    # CONSTANTS
+
+
+    # limits
+    MINIMUM_BALANCE = 10_000
+    TRANSACTION_LIMIT = 6
+
+    # fees
+    LOW_BALANCE_FEE = 100
+
+
+    ##--------------------------------------------------------------------------
+    # INSTANCE METHODS
+
+
+    attr_reader :transactions
+
+
+    def initialize(id, initial_balance)
+      if (initial_balance >= MINIMUM_BALANCE)
+        @transactions = 0
+
+        super(id, initial_balance)
+
+        return true
+
+      else
+        raise ArgumentError.new("You cannot create a money market account with less than a $#{ MINIMUM_BALANCE } initial deposit.")
+      end
+    end
+
+
+    # add interest to account.
+    def add_interest(rate_percentage)
+      if validate_number(rate_percentage)
+        # convert rate from percentage to decimal.
+        rate_decimal = rate_percentage / 100.0
+
+        # calculate interest.
+        interest = @balance * rate_decimal
+
+        # add interest to balance.
+        update_balance(interest)
+
+        # return interest.
+        return interest
+      else
+        return false
+      end
+    end
+
+
+    # deposit money to account.
+    def deposit(amount)
+      unless balance_low? # if balance is low, don't do this stuff.
+        unless transaction_allowed? # if transaction is allowed, don't send a warning message!
+          warn "You cannot make any additional deposits this month. You have already reached the limit (#{ TRANSACTION_LIMIT })."
+
+          return @balance
+        else # when it is allowed, increment transactions counter.
+          @transactions += 1
+        end
```

Comment:
You generally do not want to use unless/else blocks -- it's easy to get confused reading them. Reversing it to an if/else helps readability. 
```
if transaction_allowed?
  @transactions += 1
else
  warn "You cannot ... "
end